recipes-ni/ni-base-system-image-tests/files/fs_permissions_known.py

```diff
@@ -0,0 +1,79 @@
+import argparse
+import csv
+import os
+import sys
+import glob
+import stat
+import pwd
+import grp
+import fs_permissions_shared
+
+def log_mismatch(col, exp, act, path, logger):
+    logger.log(f'ERROR: Expected {col} \'{exp}\' but got \'{act}\' for path {path}.')
+
+class Permissions:
+    def __init__(self, file_name):
+        if not os.path.exists(file_name):
+            self.path = None
+            return
+        stats = os.lstat(file_name)
+        self.path = file_name
+        self.mode = stat.filemode(stats.st_mode)
+        self.user = pwd.getpwuid(stats.st_uid).pw_name
+        self.group = grp.getgrgid(stats.st_gid).gr_name
+        if stat.S_ISLNK(stats.st_mode):
+            self.link = os.readlink(self.path)
+        else:
+            self.link = ''
+
+def check_known_permissions(logger):
+    logger.log('###### Checking expected file permissions against actual for known folders')
```

Comment:
Looks like this is going to be the first line reported, and since it's constant the review queue won't be able to distinguish failures. Consider producing a digest and putting it into the first line with `Logger.prefix_log`.

Reply:
To expand a little on what Mike said, we have tooling that will group together similar failures across test runs. One of the things it looks at to determine whether to group failures together is the first line of the error message. If the error message is constant, it can only put all failures in one group. If the error message is dynamic, it can group more narrowly depending on the specific differences causing the failure.

That being said, it only looks at the first line and only up to the first 100 characters. So you can't cram too much information into the first line, but you can calculate a hash or checksum that uniquely identifies the failure and include that in the first line. You can look at the fs_permissions_diff test as an example.

Reply:
I think it'd be preferable to have a message based on the failure itself, rather than the version. Perhaps interleaving something like this into the check loop would be adequate:

```python
# ...
md5sum = hashlib.md5()
for path in paths:
    if not path_matches_expectations(path):
        md5sum.update(path.encode('utf-8'))
logger.prefix_log(f'known fs permissions check {md5sum.hexdigest()}')
```

(In hindsight, the fs diff test shouldn't put the version in the first line, either--that's probably a holdover from the dmesg diff test, where the kernel is upgraded at longer intervals.)